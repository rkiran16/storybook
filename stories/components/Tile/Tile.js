import { html } from "lit-html";
import { Link } from "../Link/Link";
import { CheckBox } from "../../form/CheckBox/CheckBox";

export const Tile = ({hideCheckBox = false}) => {
    return html `
        <div class="tile">
            <div class="tile__info"  style="height: 300px">
                <div class="tile__info__controls">
                    ${!hideCheckBox ? CheckBox({checked: false, showLabel: false}) : ''}
                    <div class="w-100 d-flex justify-content-end tile-control">
                        <button type="button" class="btn btn-icon btn-icon-w-text horizontal">
                            Remove
                            <span class="material-icons-outlined ms-1">delete_outline</span>
                        </button>
                    </div>
                </div>
                <div class="d-flex flex-column w-100">
                    <div class="mb-2">
                        ${Link({label:"Title Goes Here",
                        href: "http://www.cardinalhealth.com",target :"_self",
                        showIcon : false, inline: true, additionalClass: "text-break tile-title"})}
                    </div> 
                    <div class="d-flex flex-column flex-sm-row justify-content-between w-100">
                      <p>REST OF CONTENT GOES HERE
                    </div> 
                </div>
            </div>
        </div>
    `
}