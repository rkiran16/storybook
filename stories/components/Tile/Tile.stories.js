import { Tile } from "./Tile";

export default {
    title: "Components/Tiles/Tile",
    argTypes: {
        hideCheckBox: {
            name: "Hide Checkbox",
            type: { name: 'boolean', required: false},
            description: "Hide CheckBox",
            defaultValue: false,
            table: {
               type: { summary: 'boolean' },
               defaultValue: { summary: false },
            },
            control: { type: 'boolean'}
        },
    }
}

const Template = (args) => Tile(args);

export const Tile_Skeleton = Template.bind({});