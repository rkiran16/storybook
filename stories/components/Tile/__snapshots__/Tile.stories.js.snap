// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Tiles/Tile Tile Skeleton 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
        <div class=\\"tile\\">
            <div class=\\"tile__info\\"  style=\\"height: 300px\\">
                <div class=\\"tile__info__controls\\">
                    ",
    "
                    <div class=\\"w-100 d-flex justify-content-end tile-control\\">
                        <button type=\\"button\\" class=\\"btn btn-icon btn-icon-w-text horizontal\\">
                            Remove
                            <span class=\\"material-icons-outlined ms-1\\">delete_outline</span>
                        </button>
                    </div>
                </div>
                <div class=\\"d-flex flex-column w-100\\">
                    <div class=\\"mb-2\\">
                        ",
    "
                    </div> 
                    <div class=\\"d-flex flex-column flex-sm-row justify-content-between w-100\\">
                      <p>REST OF CONTENT GOES HERE
                    </div> 
                </div>
            </div>
        </div>
    ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
        "  ?disabled=",
        " id=\\"customCheck111-33\\" />
                <label class=",
        " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>
        ",
      ],
      "values": Array [
        false,
        undefined,
        "form-check-label visually-hidden",
      ],
    },
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<a href=",
        " class=",
        " target=",
        ">",
        "</a>",
      ],
      "values": Array [
        "http://www.cardinalhealth.com",
        "inline text-break tile-title",
        "_self",
        "Title Goes Here",
      ],
    },
  ],
}
`;
